.github/workflows/build_and_test.yml

```diff
@@ -0,0 +1,287 @@
+
+      - name: Verify files
+        run: |
+          curl -sSL https://secchannel.rsk.co/SUPPORT.asc | gpg2 --import -
+          gpg2 --verify SHA256SUMS.asc && sha256sum --check SHA256SUMS.asc
+
+      - uses: actions/cache@v4
+        name: Cache Gradle
+        id: cache-gradle
+        with:
+          path: |
+            .gradle/caches
+            gradle/wrapper
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+
+      - name: Get gradle wrapper and build dependencies
+        run: |
+          ./configure.sh
+          ./gradlew --no-daemon dependencies
+
+      - name: Build
+        run: |
+          ./gradlew --no-daemon --stacktrace build -x test
+
+      - name: Archive build artifacts
+        uses: actions/upload-artifact@v4
+        with:
+          name: build-files
+          path: |
+            rskj-core/build
+
+  smell-test:
+    needs: prepare-rskj
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Setup Java JDK
+        uses: actions/setup-java@v3
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+
+      - uses: actions/cache@v4
+        name: Restore Gradle cache
+        id: cache-gradle
+        with:
+          path: |
+            .gradle/caches
+            gradle/wrapper
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+
+      - name: Get gradle wrapper and build dependencies
+        run: |
+          if [ ! -f gradle/wrapper/gradle-wrapper.jar ]; then
+            ./configure.sh
+            ./gradlew --no-daemon dependencies
+          fi
+
+      - name: Download build artifacts
+        uses: actions/download-artifact@v4
+        with:
+          name: build-files
+          path: |
+            rskj-core/build
+
+      - name: If GH event is PR
+        if: github.event_name == 'pull_request'
+        run: |
+          git branch -f master origin/master
+          pr_number==${{ github.event.pull_request.number }}
+          extra_flags="-Dsonar.pullrequest.base=master
+            -Dsonar.pullrequest.branch=${{ github.head_ref }}
+            -Dsonar.pullrequest.key=$pr_number"
+          ./gradlew sonarqube --no-daemon -x build -x test \
+            $extra_flags \
+            -Dsonar.organization=rsksmart \
+            -Dsonar.host.url="https://sonarcloud.io" \
+            -Dsonar.token="${{ secrets.SONAR_TOKEN }}"
+
+      - name: If not
+        if: github.event_name != 'pull_request'
+        run: |
+          ./gradlew sonarqube --no-daemon -x build -x test \
+            -Dsonar.branch.name=master \
+            -Dsonar.organization=rsksmart \
+            -Dsonar.host.url="https://sonarcloud.io" \
+            -Dsonar.token="${{ secrets.SONAR_TOKEN }}"
```

Comment:
actually it is in the GHA UI where it makes sense as you always see the skipped steps.
but fair enough, I can change it to something more descriptive

Reply:
> Is there a particular reason for spliting it at the step level, instead of doing everything in a single step and using shell conditionals?

yes, https://github.com/rsksmart/rskj/pull/2290#discussion_r1626308422